common/djangoapps/third_party_auth/tests/test_strategy.py

```diff
@@ -95,3 +97,65 @@ def test_sets_email_if_not_provided(self, email, fallback_domain, expected_email
 
         _, user_data = self._get_last_call_args(patched_create_account)
         self.assertEqual(user_data['email'], expected_email)
+
+    @ddt.data(
+        (None, 'host', 'host'),
+        ("", 'other_host', 'other_host'),
+        ('x_forwarded_host', 'irrelevant', 'x_forwarded_host'),
+        ('other_x_forwarded_host', 'still_irrelevant', 'other_x_forwarded_host'),
+    )
+    @ddt.unpack
+    @override_settings(RESPECT_X_FORWARDED_HEADERS=True)
+    def test_request_host(self, x_forwarded_value, get_host_value, expected_value, unused_patch):
+        self.request_mock.META = {}
+        self.request_mock.get_host.return_value = get_host_value
+        if x_forwarded_value is not None:
+            self.request_mock.META['HTTP_X_FORWARDED_HOST'] = x_forwarded_value
+
+        self.assertEqual(self.strategy.request_host(), expected_value)
+
+    @ddt.data(
+        (None, 'port', 'port'),
+        ("", 'other_port', 'other_port'),
+        ('x_forwarded_port', 'irrelevant', 'x_forwarded_port'),
+        ('other_x_forwarded_port', 'still_irrelevant', 'other_x_forwarded_port'),
+    )
+    @ddt.unpack
+    @override_settings(RESPECT_X_FORWARDED_HEADERS=True)
+    def test_request_port(self, x_forwarded_value, server_port_value, expected_value, unused_patch):
+        self.request_mock.META = {'SERVER_PORT': server_port_value}
+        if x_forwarded_value is not None:
+            self.request_mock.META['HTTP_X_FORWARDED_PORT'] = x_forwarded_value
+
+        self.assertEqual(self.strategy.request_port(), expected_value)
+
+    @ddt.data(
+        (None, 'host', 'host'),
+        ("", 'other_host', 'other_host'),
+        ('x_forwarded_host', 'normal_host', 'normal_host'),
+        ('other_x_forwarded_host', 'other_normal_host', 'other_normal_host'),
+    )
+    @ddt.unpack
+    @override_settings(RESPECT_X_FORWARDED_HEADERS=False)
+    def test_request_host_ignore_x_forwarded(self, x_forwarded_value, get_host_value, expected_value, unused_patch):
+        self.request_mock.META = {}
+        self.request_mock.get_host.return_value = get_host_value
+        if x_forwarded_value is not None:
+            self.request_mock.META['HTTP_X_FORWARDED_HOST'] = x_forwarded_value
+
+        self.assertEqual(self.strategy.request_host(), expected_value)
```

Comment:
This test body is identical to `test_request_host`. Is there a clean way to combine them?